orchagent/montxorch.cpp

```diff
@@ -0,0 +1,251 @@
+#include <linux/if_ether.h>
+
+#include <unordered_map>
+#include <utility>
+#include <exception>
+#include <vector>
+
+#include "montxorch.h"
+#include "orch.h"
+#include "port.h"
+#include "logger.h"
+#include "sai_serialize.h"
+#include "converter.h"
+#include "portsorch.h"
+
+extern sai_port_api_t *sai_port_api;
+extern PortsOrch*       gPortsOrch;
+using namespace std::rel_ops;
+
+const std::string currentDateTime() 
+{
+    time_t     now = time(0);
+    struct tm  tstruct;
+    char       buf[80];
+    tstruct = *localtime(&now);
+    strftime(buf, sizeof(buf), "%Y-%m-%d.%X", &tstruct);
+    return buf;
+}
+
+MonTxOrch::MonTxOrch(TableConnector appDb, TableConnector confDb, TableConnector stateDb):
+    Orch(confDb.first,confDb.second),
+    m_TxErrorTable(appDb.first,appDb.second),
+    m_stateTxErrorTable(stateDb.first,stateDb.second),
+    m_pollPeriod(0)
+{
+    SWSS_LOG_ENTER();    
+    m_pollTimer = new SelectableTimer(timespec { .tv_sec = 0, .tv_nsec = 0 });
+    Orch::addExecutor( new ExecutableTimer(m_pollTimer, this, TXPORTMONORCH_SEL_TIMER));
+    SWSS_LOG_NOTICE("MonTxOrch initialized with the following tables %s %s \n",stateDb.second.c_str(),appDb.second.c_str());
+}
+
+void MonTxOrch::startTimer(uint32_t interval)
+{
+    SWSS_LOG_ENTER();
+    try
+    {
+        auto timespecInterval = timespec { .tv_sec = interval , .tv_nsec = 0 };
+        SWSS_LOG_INFO("MonTxOrch::startTimer with executor: %p\n", m_pollTimer);
+        m_pollPeriod = interval;
+        m_pollTimer->setInterval(timespecInterval);
+        m_pollTimer->stop();
+        m_pollTimer->start();
+    }
+    catch (...)
+    {
+        SWSS_LOG_ERROR("MonTxOrch::startTimer function failure\n");
+    }
+}
+
+int MonTxOrch::periodUpdateHandler(const vector<FieldValueTuple>& data)
+{
+    bool restart = false;
+    uint32_t newPollPeriod = 0;
+    SWSS_LOG_ENTER();
+    for (auto iter_ : data)
+    {
+        try 
+        { 
+            if (fvField(iter_) == MONTXORCH_CFG_PERIOD)
+            {
+                
+                newPollPeriod =  static_cast<uint32_t>(stoul(fvValue(iter_)));
+                restart |= (m_pollPeriod != newPollPeriod);  //Differenc in current period and existing poll period causes restart
+                SWSS_LOG_INFO("MonTxOrch::periodUpdateHandler Update Monitoring Period to %u\n", newPollPeriod);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("MonTxOrch::periodUpdateHandler Invalid field type provided: %s\n", fvField(iter_).c_str());
+                return -1;
+            }
+        }
+        catch (...) 
```

Comment:
What exception are you trying to catch here? Which line do you expect to throw an exception?